python/private/pypi/env_marker_setting.bzl

```diff
@@ -0,0 +1,265 @@
+"""Implement a flag for matching the dependency specifiers at analysis time."""
+
+load("@bazel_skylib//rules:common_settings.bzl", "BuildSettingInfo")
+load("//python/private:toolchain_types.bzl", "TARGET_TOOLCHAIN_TYPE")
+load(":pep508_evaluate.bzl", "evaluate")
+
+# todo: copied from pep508_env.bzl
+_os_name_select_map = {
+    # The "java" value is documented, but with Jython defunct,
+    # shouldn't occur in practice.
+    # The os.name value is technically a property of the runtime, not the
+    # targetted runtime OS, but the distinction shouldn't matter if
+    # things are properly configured.
+    "@platforms//os:windows": "nt",
+    "//conditions:default": "posix",
+}
+
+# TODO @aignas 2025-04-29: this is copied from ./pep508_env.bzl
+_platform_machine_aliases = {
+    # These pairs mean the same hardware, but different values may be used
+    # on different host platforms.
+    "amd64": "x86_64",
+    "arm64": "aarch64",
+    "i386": "x86_32",
+    "i686": "x86_32",
+}
+
+# Taken from
+# https://docs.python.org/3/library/sys.html#sys.platform
+_sys_platform_select_map = {
+    # These values are decided by the sys.platform docs.
+    "@platforms//os:android": "android",
+    "@platforms//os:emscripten": "emscripten",
+    "@platforms//os:ios": "ios",
+    "@platforms//os:linux": "linux",
+    "@platforms//os:osx": "darwin",
+    "@platforms//os:windows": "win32",
+    "@platforms//os:wasi": "wasi",
+    # NOTE: The below values are approximations. The sys.platform() docs
+    # don't have documented values for these OSes. Per docs, the
+    # sys.platform() value reflects the OS at the time Python was *built*
+    # instead of the runtime (target) OS value.
+    "@platforms//os:freebsd": "freebsd",
+    "@platforms//os:openbsd": "openbsd",
```

Comment:
If I am reading it correctly, this value might be `freebsd8` and I am wondering if we should do some trickery to combine this value with the `platform_release`?